ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -395,31 +388,22 @@ private boolean hasConflictWithCache(Collection<Account> accountsToSet) {
   private boolean hasConflictingAccount(Collection<Account> accountsToSet, AccountInfoMap accountInfoMap) {
     boolean res = false;
     for (Account account : accountsToSet) {
+      // if the account already exists, check that the snapshot version matches the expected value.
       Account accountInMap = accountInfoMap.getAccountById(account.getId());
-
-      // @todo if accountInMap is not null, make sure
-      // @todo account.generationId() = accountInfoMap.getById(account.getId()).generationId()+1
-      // @todo the generationId will be added to Account class.
-
-      // check for case D described in the javadoc of AccountService
-      if (accountInMap == null && accountInfoMap.containsName(account.getName())) {
-        Account conflictingAccount = accountInfoMap.getAccountByName(account.getName());
+      if (accountInMap != null && account.getSnapshotVersion() != accountInMap.getSnapshotVersion()) {
         logger.error(
-            "Account to update with accountId={} accountName={} conflicts with an existing record with accountId={} accountName={}",
-            account.getId(), account.getName(), conflictingAccount.getId(), conflictingAccount.getName());
+            "Account to update (accountId={} accountName={}) has an unexpected snapshot version in zk (expected={}, encountered={})",
+            account.getId(), account.getName(), account.getSnapshotVersion(), accountInMap.getSnapshotVersion());
         res = true;
```

Comment:
I will add the break back in. The only downside of breaking is that only the first bad account will be mentioned in the logs.

Reply:
That's a fair argument to make - it might be useful to see all the clashes. If there are no other repercussions, then you don't have to put the `break` back in.